File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
    
        int n=nums.size();
        if(n==0)return 0;
        set<int> st;
        for(int ele:nums){
            st.insert(ele);
        }
        int gAns=0;
        int lAns=0;
        int prev=*st.begin()-1;
        for(int ele : st){
            if(ele==prev+1){
                lAns++;
            }
            else{
                lAns=1;
            }
            gAns=max(gAns,lAns);
            prev=ele;
        }
        return gAns;
        
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    nums=nums.sort((a,b)=>{
        return a-b;
    });
    const myset=new Set();
    nums.forEach((ele)=>{
        myset.add(ele);
    });
    let prev=myset[0]-1;
    let lans=0,ans=0;
    for(let ele of myset){
        if(ele==prev+1){
            lans++;
        }
        else{
            lans=1;
        }
        ans=Math.max(ans,lans);
        prev=ele;
    }
    return ans;
    
};